File: Go/1431. Kids With the Greatest Number of Candies.go
package main

import "slices"

//goland:noinspection GoUnusedFunction
func kidsWithCandies(candies []int, extraCandies int) []bool {
	maxCandies := slices.Max(candies)

	candiesWithExtra := make([]bool, len(candies))

	for i, candy := range candies {
		candiesWithExtra[i] = candy+extraCandies >= maxCandies
	}

	return candiesWithExtra
}

File: TypeScript/1431. Kids With the Greatest Number of Candies.ts
// noinspection JSUnusedGlobalSymbols
const kidsWithCandies = (candies: number[], extraCandies: number): boolean[] => {
  const max: number = Math.max(...candies)
  return candies.map((item: number): boolean => item + extraCandies >= max)
}
